src/App.js

```diff
@@ -183,7 +183,7 @@ class OldApp extends Component {
     const network = tx.chainId ? ethereumUtils.getNetworkFromChainId(tx.chainId) : tx.network || networkTypes.mainnet;
     const isL2 = isL2Network(network);
 
-    const provider = getCachedProviderForNetwork(network);
+    const provider = getProviderForNetwork(network);
     const providerUrl = provider?.connection?.url;
     const connectedToHardhat = isHardHat(providerUrl);
```

Comment:
Think this one can just be:

```suggestion
    const connectedToHardhat = getIsHardhatConnected();
```

Reply:
Good call!